src/components/Form/FormFields/PhoneNumberFormField.tsx

```diff
@@ -17,157 +16,148 @@ import {
-      }
-      setCountry(phoneCodes[getCountryCode(field.value)!]);
-    }
-  }, [setValue]);
+    }, [error]);
+    const handleCountryChange = (value: CountryData): void => {
+      setCountry(value);
+      setValue(conditionPhoneCode(value.code));
+    };
 
-  return (
-    <FormField
-      field={{
-        ...field,
-        error: field.error ?? error,
-        labelSuffix: field.labelSuffix || (
-          <PhoneNumberTypesHelp types={props.types} />
-        ),
-      }}
-    >
-      <div className="relative rounded-md shadow-sm">
-        <Popover>
-          {({ open }: { open: boolean }) => {
-            return (
-              <>
-                <PopoverButton className="absolute h-full">
-                  <div className="absolute inset-y-0 left-0 m-0.5 flex w-[4.5rem] cursor-pointer items-center justify-around bg-slate-100">
-                    <span className="rounded-md pl-4">
-                      {country?.flag ?? "🇮🇳"}
-                    </span>
-                    <CareIcon
-                      icon="l-angle-down"
-                      className={`text-2xl font-bold ${open && "rotate-180"}`}
-                    />
-                  </div>
-                </PopoverButton>
-                <input
-                  type="tel"
-                  id={field.id}
-                  name={field.name}
-                  autoComplete={props.autoComplete ?? "tel"}
-                  className={classNames(
-                    "cui-input-base h-full pl-20 tracking-widest sm:leading-6",
-                    field.error && "border-danger-500",
-                    field.className,
-                  )}
-                  maxLength={field.value?.startsWith("1800") ? 11 : 15}
-                  placeholder={props.placeholder}
-                  value={formatPhoneNumber(field.value, props.types)}
-                  onChange={(e) => setValue(e.target.value)}
-                  disabled={field.disabled}
-                  onBlur={() => setError(validate(field.value, "blur"))}
-                />
-                <PopoverPanel className="w-full">
-                  {({ close }) => (
-                    <CountryCodesList
-                      handleCountryChange={handleCountryChange}
-                      onClose={close}
-                    />
-                  )}
-                </PopoverPanel>
-              </>
-            );
-          }}
-        </Popover>
-      </div>
-    </FormField>
-  );
-}
+    useEffect(() => {
+      if (field.value && field.value.length > 0) {
+        if (field.value.startsWith("1800")) {
+          setCountry({ flag: "📞", name: "Support", code: "1800" });
+          return;
+        }
+        if (field.value === "+") {
+          setCountry({ flag: "🌍", name: "Other", code: "+" });
+          return;
+        }
+        setCountry(phoneCodes[getCountryCode(field.value)!]);
+      }
+    }, [setValue]);
 
-const PhoneNumberTypesHelp = (props: { types: PhoneNumberType[] }) => {
```

Comment:
Why was `PhoneNumberTypesHelp` component removed? It is needed in other places?

When making a change to a reusable component like this, make sure it does not affect existing usages in other places.

Reply:
> Why was `PhoneNumberTypesHelp` component removed? It is needed in other places?
> 
> When making a change to a reusable component like this, make sure it does not affect existing usages in other places.

@rithviknishad ,it is not used in that file may be used any where else but if I add this it will throw eslint error

Reply:
it is used; u just removed the usage too...

<img width="781" alt="image" src="https://github.com/user-attachments/assets/49b31f1b-1866-41e9-95ff-44616c8d206d">

Reply:
> it is used; u just removed the usage too...
> 
> <img alt="image" width="781" src="https://private-user-images.githubusercontent.com/25143503/381665807-49b31f1b-1866-41e9-95ff-44616c8d206d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAzMTIyOTYsIm5iZiI6MTczMDMxMTk5NiwicGF0aCI6Ii8yNTE0MzUwMy8zODE2NjU4MDctNDliMzFmMWItMTg2Ni00MWU5LTk1ZmYtNDQ2MTZjOGQyMDZkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDMwVDE4MTMxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFjNjBmYTVhYjJmYWRkOWEwZGNmYmMzYzBkNjNiYWU5YjhmMDZmYzExNTRjZDVmYTJmOTEwZWY3NjBmOWUzMDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.JR_z_v7LHiVwZTdwzr-oTVFqhcuCX9slndcm_T7KYCQ">

@rithviknishad I have not done this after merging bodhis's branch it happen ,and after adding this I found a bad breaking ui 
![Screenshot 2024-10-30 235503](https://github.com/user-attachments/assets/29102651-a443-4b9a-8ba6-a1e548d1a0e4)

Reply:
we could conditionally hide it only for the search thing;

Reply:
@rithviknishad yeh I saw in the console it is coming only for phnuberinput and emergencyphnuminput I can use condition for this two only

Reply:
@rithviknishad ,yah only for the searchbymultifield compnent which is newly build,I am passing a props help={fasle} and for this compnent 
heplcomponenet means this " ? " will  be hidden and for other compnent it will be true .for this I used useffect and usestate inside formfield to check . Also declared the type help?boolean ,and it is working properly